packages/medusa/src/services/event-bus.ts

```diff
@@ -91,115 +110,120 @@ class EventBusService {
 
   /**
    * Adds a function to a list of event subscribers.
-   * @param {string} event - the event that the subscriber will listen for.
-   * @param {func} subscriber - the function to be called when a certain event
+   * @param event - the event that the subscriber will listen for.
+   * @param subscriber - the function to be called when a certain event
    * happens. Subscribers must return a Promise.
+   * @return this
    */
-  subscribe(event, subscriber) {
+  subscribe(event: string | symbol, subscriber: Subscriber): this {
     if (typeof subscriber !== "function") {
       throw new Error("Subscriber must be a function")
     }
 
-    if (this.observers_[event]) {
-      this.observers_[event].push(subscriber)
-    } else {
-      this.observers_[event] = [subscriber]
-    }
+    const observers = this.observers_.get(event) ?? []
+    this.observers_.set(event, [...observers, subscriber])
+
+    return this
   }
 
   /**
    * Adds a function to a list of event subscribers.
-   * @param {string} event - the event that the subscriber will listen for.
-   * @param {func} subscriber - the function to be called when a certain event
+   * @param event - the event that the subscriber will listen for.
+   * @param subscriber - the function to be called when a certain event
    * happens. Subscribers must return a Promise.
+   * @return this
    */
-  unsubscribe(event, subscriber) {
+  unsubscribe(event: string | symbol, subscriber: Subscriber): this {
     if (typeof subscriber !== "function") {
       throw new Error("Subscriber must be a function")
     }
 
-    if (this.observers_[event]) {
-      const index = this.observers_[event].indexOf(subscriber)
+    if (this.observers_.get(event)?.length) {
+      const index = this.observers_.get(event)?.indexOf(subscriber)
       if (index !== -1) {
         this.observers_[event].splice(index, 1)
```

Comment:
question: We've changed `this.observers_` to be a `Map` instead of an array, so I would assume, that we cannot access an element by index as done here, but please correct me if I am wrong. Should it not be `this.observers_.delete(event)`?

Reply:
Ah yeah, makes sense! 😅